### PR TITLE
INFENG-6249 Harbour to Kubernetes Migration

### PR DESCRIPTION
**Pull request by tylerstapler-wf**
# Harbour to Kubernetes Migration

This PR should contain everything you need to migrate your service to Kubernetes.

We have automatically generated as much of the updates as possible, but we will need your cooperation to make the migration successful.

The TODO's section below contains the steps you will need in order to complete your Helm chart migration. You can also find the TODOs in the `README.md` within the Helm chart (`helm/slate/README.md#todos`) In addition to the TODOs, `README.md` contains helpful explainations of the chart's components.

When you're ready to test the Helm chart migration, [see our development documentation](https://github.com/Workiva/k8s-local/blob/master/docs/local_dev.md).

When you think you're ready to merge your PR see our [final steps documentation](https://github.com/Workiva/k8s-local/blob/master/docs/final_migration.md).

For additional help, please see the hipchat room "*Support: Kubernetes Migration*".

@tylerstapler-wf @kevinfrommelt-wf

## TODO's

- [ ] Verify that there aren't any sensitive values in environment section. Clean up unused secrets and environment vars in `values.yaml`
- [ ] Configure Liveness probe. See `values.yaml`
- [ ] Enable CPU or memory based autoscaling for your application if possible. See `values.yaml`
- [ ] If possible, raise your `minAvailable` and `autoscaling.minReplicas` to at least three for your application.
  This is a best practice which helps to ensure a highly available application.
- [ ] Check if your container can run as non-root user and update values.yaml accordingly
- [ ] Upgrade to latest app-intelligence/messaging-sdk libraries.
- [ ] Update your team name in the charts.yaml.
- [ ] Select a serviceAvailabilityType in the charts.yaml.
- [ ] Update the value `image.tag` in `values.yaml` to match the image produced by this PR. It should contain the necessary App Intelligence and Messaging SDK updates required to run in Kubernetes.
- [ ] Check that all of the ports you are listening on are configured in your `service.yaml`

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on HipChat: InfoSec Forum._

**Comment by ericlarssen-wf** [comment]
@dustinhiatt-wf although it looks like Tyler opened this PR, I will be working with you to getting this merged and out to wk-dev

**Comment by dustinhiatt-wf** [comment]
@ericlarssen-wf

**Comment by charliestrawn-wf** [comment]
This is running in wk-dev https://rancher.workiva.net/p/c-ncmr5:p-4prw9/workload/deployment:workiva:cerebral-docs and accessible here https://k.wk-dev.wdesk.org/s/cerebral-docs (notice k.wk-dev instead of h.)

**Comment by dustinhiatt-wf** [comment]
+1

File: helm/cerebral-docs/values.yaml [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  # - type: cpu
+  #   average_percent: 75
+  # - type: memory
+  #   average_percent: 75
+
+# Ingress configuration
+# See: https://kubernetes.io/docs/concepts/services-networking/ingress/
+ingress:
+  # Base domain for the cluster (ex. wk-dev.wdesk.org, app.wdesk.org).
+  clusterDomain: 'wk-dev.wdesk.org'
```
Comment (charliestrawn-wf) [inline helm/cerebral-docs/values.yaml]: we've been softly encouraging teams to leave this as an empty string, as it will be passed in via RM pipelines

**Comment by dustinhiatt-wf** [comment]
+1

**Comment by charliestrawn-wf** [comment]
+1

**Comment by dustinhiatt-wf** [comment]
QA +1

**Pull request merged**